gcloud-java-storage/src/main/java/com/google/gcloud/spi/StorageRpc.java

```diff
@@ -132,6 +133,89 @@ public BatchResponse(Map<StorageObject, Tuple<Boolean, StorageException>> delete
     }
   }
 
+  class RewriteRequest {
+
+    public final StorageObject source;
+    public final Map<StorageRpc.Option, ?> sourceOptions;
+    public final StorageObject target;
+    public final Map<StorageRpc.Option, ?> targetOptions;
+    public final Long megabytesRewrittenPerCall;
+
+    public RewriteRequest(StorageObject source, Map<StorageRpc.Option, ?> sourceOptions,
+        StorageObject target, Map<StorageRpc.Option, ?> targetOptions,
+        Long megabytesRewrittenPerCall) {
+      this.source = source;
+      this.sourceOptions = sourceOptions;
+      this.target = target;
+      this.targetOptions = targetOptions;
+      this.megabytesRewrittenPerCall = megabytesRewrittenPerCall;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == null) {
+        return false;
+      }
+      if (!(obj instanceof RewriteRequest)) {
+        return false;
+      }
+      final RewriteRequest other = (RewriteRequest) obj;
+      return Objects.equals(this.source, other.source)
+          && Objects.equals(this.sourceOptions, other.sourceOptions)
+          && Objects.equals(this.target, other.target)
+          && Objects.equals(this.targetOptions, other.targetOptions)
+          && Objects.equals(this.megabytesRewrittenPerCall, other.megabytesRewrittenPerCall);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(source, sourceOptions, target, targetOptions, megabytesRewrittenPerCall);
+    }
+  }
+
+  class RewriteResponse {
+
+    public final RewriteRequest rewriteRequest;
+    public final StorageObject result;
+    public final Long blobSize;
+    public final Boolean isDone;
+    public final String rewriteToken;
+    public final Long totalBytesRewritten;
+
+    public RewriteResponse(RewriteRequest rewriteRequest, StorageObject result, Long blobSize,
```

Comment:
Any reason `blobSize`, `isDone` and `totalBytesRewritten` are not primitives? Can they be `null` after the first call?